File: represent/bench/java_flat/FormalBench_0238__FormalBenchHF__diverse__rule_1__IsNumDecagonal_llm_2.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class IsNumDecagonal {

	public static int isNumDecagonal(int number) {
		int[] arr = new int[number + 1];
		arr[0] = 0;
		for (int c = 0; c <= number - 1; c++) {
			arr[c + 1] = arr[c] + 8 * c + 1;
		}
		return arr[number];
	}
}

File: represent/bench/c_flat/FormalBench_0238__FormalBenchHF__diverse__rule_1__IsNumDecagonal_llm_2.c
#include <stdlib.h>

int isNumDecagonal(int number);

int isNumDecagonal(int number) {

		int *arr = (int *)malloc(sizeof(int) * (number + 1));
int arr_len = number + 1;
		arr[0] = 0;
		for (int c = 0; c <= number - 1; c++) {
			arr[c + 1] = arr[c] + 8 * c + 1;
		}
		return arr[number];
}
